pkg/deploy/controller/deployment/controller.go

```diff
@@ -289,6 +290,14 @@ func (c *DeploymentController) makeDeployerPod(deployment *kapi.ReplicationContr
 
 	gracePeriod := int64(10)
 
+	// In case the deployment config strategy does not set the resource requests,
+	// we default them.
+	if deploymentConfig.Spec.Strategy.Resources.Requests == nil {
```

Comment:
@derekwaynecarr I think the initial goal was to workaround the missing LimitRange (my fault) and this is now not longer needed (@kargakis?)

Yes, you will have to manually override the resources on strategy to run with BestEffort. That might break the backward compatibility right? I think that is one more reason to close this for now.